scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -437,11 +436,6 @@ int dom_xfer_task(u64 tptr, u32 new_dom_id, u64 now)
 
 	p = tptr_to_task(tptr);
 
-	if (!p) {
```

Comment:
I don't think removing this makes sense. The only place `dom_xfer_task‎` is called is in `task_set_domain`, which already does the lookup of the task. So the `struct task_struct *p` should probably just be passed as a parameter to this function along with the `struct task_ctx`. cc @likewhatevs

Reply:
> I don't think removing this makes sense. The only place dom_xfer_task‎ is called is in task_set_domain, which already does the lookup of the task. So the struct task_struct *p should probably just be passed as a parameter to this function along with the struct task_ctx.

If that passes the verifier, I think that makes the most sense. That being said, p has always been an external lookup (previously `bpf_task_from_pid`, now `bpf_probe_read_kernel`), and IIRC/IIUC that's key to getting this bit of code passing the verifier (at least it was when I last edited it).

WRT/ this change as it is now, I think it works and works fine *but* it's kinda relying on `lookup_task_ctx` to throw an error to do the right thing (which it does now), but I'm not super sure that reliance is a great thing (i.e. better to explicitly check and throw for when that changes IMO).

I guess I'm not super opposed to this change as is if like there's a reason (like idk making things run on aarch64 or something) for it, but idk of any such reason atm. 

@vax-r , what compilation error is this fixing (i.e. CI passes without this)?

Reply:
@likewhatevs -
sorry for the late reply, I'm not sure why some compiling errors and warnings keep popping up while CI doesn't show those warnings. For the part I'm trying to fix , the compilation message is `src/bpf/main.bpf.c:430:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]`, after I execute `meson compile -C build` from scx root directory.

Reply:
I recommend going with the approach @hodgesds recommended or a way without the downsides I mentioned above (because these issues aren't on CI yet).

Reply:
> I recommend going with the approach @hodgesds recommended or a way without the downsides I mentioned above (because these issues aren't on CI yet).

No problem , I understand you after some further study and I restore the part of the checking within `dom_xfer_task()` , task a look while you're available, thanks.